### PR TITLE
feat(bpdm): add bpdm orchestrator roles and svc acc

### PR DESCRIPTION
**Pull request by typecastcloud**
## Description

Add following BPDM technical user roles for CX Operator:

- BPDM Orchestrator Admin
- BPDM Orchestrator Task Creator
- BPDM Orchestrator Processor Clean
- BPDM Orchestrator Processor CleanAndSync
- BPDM Orchestrator Processor PoolSync

Add BPDM service accounts for CX Operator:

- sa-cl7-cx-1
- sa-cl7-cx-7
- sa-cl25-cx-1
- sa-cl25-cx-2
- sa-cl25-cx-3

## Why

eclipse-tractusx/portal-iam#155 added new BPDM roles and service accounts. Adding these in seeding data allows for configuration via portal.

## Issue

eclipse-tractusx/portal-backend#1020

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ ] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [ ] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [ ] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by typecastcloud** [comment]
Added test cases in issue:
https://github.com/eclipse-tractusx/portal-backend/issues/1020#issuecomment-2382674257

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/identities.json [hunk 1]
```diff
@@ -120,5 +120,45 @@
...
     "user_entity_id": "e8bc6470-28ee-4c40-a2d9-27c6e78f303b",
     "identity_type_id": 2
+  },
+  {
+    "id": "48fb21ad-4439-455b-92bb-8c54801e3eab",
+    "company_id": "2dc4249f-b5ca-4d42-bef1-7a7a950a4f87",
+    "date_created": "2024-09-27 10:01:33.439000 +00:00",
+    "user_status_id": 1,
+    "user_entity_id": "obsolete-obso-lete-obso-obsolete0001",
+    "identity_type_id": 2
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Seeder/Data/identities.json:130]: UserEntityId is nullable:
```suggestion
    "user_status_id": 1,
    "user_entity_id": null,
    "identity_type_id": 2
```

File: src/framework/Framework.Async/Directory.Build.props [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 <Project>
   <PropertyGroup>
-    <VersionPrefix>2.10.0</VersionPrefix>
+    <VersionPrefix>2.11.0</VersionPrefix>
```
Comment (ntruchsess) [inline src/framework/Framework.Async/Directory.Build.props:22]: this change seems to be redundant. There seem to be no code-changes in framework modules

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/framework/Framework.Async/Directory.Build.props (hunk 2 above)
Comment (typecastcloud) [reply src/framework/Framework.Async/Directory.Build.props:22]: Hi @ntruchsess, how do I use the update script to only update relevant versions?

Comment (typecastcloud) [reply src/framework/Framework.Async/Directory.Build.props]: Reverted. Mixed up PR.

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json [hunk 3]
```diff
@@ -124,5 +124,45 @@
     "company_service_account_kind_id": 1,
     "client_id": "beb01d13-04e2-4a2b-a909-8b4166b3dcf7",
     "client_client_id": "sa-cl2-05"
+  },
+  {
+    "id": "48fb21ad-4439-455b-92bb-8c54801e3eab",
+    "name": "sa-cl7-cx-1",
+    "description": "Client for the BPDM Portal Gate to consume golden record data from the Pool.",
```
Comment (evegufy) [inline src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json:131]:
```suggestion
    "description": "Technical User for the BPDM Portal Gate to consume golden record data from the Pool.",
```

I'll make sure to also update the description for all new service accounts in portal-iam, didn't notice it during the review of https://github.com/eclipse-tractusx/portal-iam/pull/155

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json [hunk 4]
```diff
@@ -124,5 +124,45 @@
...
+    "name": "sa-cl7-cx-7",
+    "description": "Technical User for BPDM services to communicate between each other to realize the golden record process: used by the Portal Gate, Pool and Cleaning Service.",
+    "company_service_account_type_id": 2,
+    "company_service_account_kind_id": 1,
+    "client_client_id": "sa-cl7-cx-7"
+  },
+  {
+    "id": "d33043a0-ed8f-46b7-9319-8f3f6f052c28",
+    "name": "sa-cl25-cx-1",
+    "description": "Client for the BPDM cleaning service dummy component to process golden record tasks from the Orchestrator.",
```
Comment (evegufy) [inline src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json:147]:
```suggestion
    "description": "Technical User for the BPDM cleaning service dummy component to process golden record tasks from the Orchestrator.",
```

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json [hunk 5]
```diff
@@ -124,5 +124,45 @@
...
+    "company_service_account_type_id": 2,
+    "company_service_account_kind_id": 1,
+    "client_client_id": "sa-cl25-cx-1"
+  },
+  {
+    "id": "fc729705-adf0-499e-91c7-1e04d817bad8",
+    "name": "sa-cl25-cx-2",
+    "description": "Client for the BPDM Pool component to process golden record tasks from the Orchestrator.",
```
Comment (evegufy) [inline src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json:155]:
```suggestion
    "description": "Technical User for the BPDM Pool component to process golden record tasks from the Orchestrator.",
```

File: src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json [hunk 6]
```diff
@@ -124,5 +124,45 @@
...
+    "company_service_account_type_id": 2,
+    "company_service_account_kind_id": 1,
+    "client_client_id": "sa-cl25-cx-2"
+  },
+  {
+    "id": "c8ab54ea-830e-4f8a-a540-336e0576a619",
+    "name": "sa-cl25-cx-3",
+    "description": "Client for the BPDM Portal Gate to create and monitor golden record tasks inside the Orchestrator.",
```
Comment (evegufy) [inline src/portalbackend/PortalBackend.Migrations/Seeder/Data/company_service_accounts.json:163]:
```suggestion
    "description": "Technical User for the BPDM Portal Gate to create and monitor golden record tasks inside the Orchestrator.",
```

**Review by evegufy — CHANGES_REQUESTED** [review]
Hi @typecastcloud thank you for the PR, awesome!
could you please rebase to main? IMO, that will solve the issue with the currently failing build https://github.com/eclipse-tractusx/portal-backend/actions/runs/11139606243/job/30956625767?pr=1048
could you please also change the PR title to `feat(bpdm): add bpdm orchestrator roles and svc acc` to follow conventional commits?

**Comment by typecastcloud** [comment]
> Hi @typecastcloud thank you for the PR, awesome! could you please rebase to main? IMO, that will solve the issue with the currently failing build https://github.com/eclipse-tractusx/portal-backend/actions/runs/11139606243/job/30956625767?pr=1048 could you please also change the PR title to `feat(bpdm): add bpdm orchestrator roles and svc acc` to follow conventional commits?

Title renamed and rebased. Still the same build issue.

When I run tests, order for `/api/administration/registration/Network/companies` in `docs/api/administration-service.yaml` changes.

**Review by evegufy — APPROVED** [review]
lgtm
@ntruchsess could you please review again?

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**